processing/src/main/java/io/druid/segment/ObjectColumnSelector.java

```diff
@@ -68,12 +68,23 @@ default double getDouble()
    */
   @Deprecated
   @Override
-  default long getLong()
+  public long getLong()
   {
     T value = getObject();
     if (value == null) {
       return 0;
```

Comment:
I think it should throw exception, if NullHandling.sqlCompatible() is true, but if this selector is treated as e. g. long-typed selector, isNull() always returns false, the user doesn't expect getLong() to throw an exception then. Isn't there an incoherence in the API system that we built?

Reply:
Well, the initial API was coherent in the sense that isNull implementation for ObjectColumnSelector used to return meaningful true/false depending on whether the object is null or not. It was changed based on your feedback to not allow users to call for optimizing the case when getObject() is heavy.

Since the ObjectColumnSelectors can be treated as object/long/double/float column, it would make sense to revert to initial implementation and return meaningful true/false from isNull for ObjectColumnSelectors. 

does this sound good to you ?

Reply:
I don't like both. It's important to get it right. I will return when I will review this PR until the end, ignore this comment for now.